.github/workflows/image.yaml

```diff
@@ -63,28 +62,26 @@ jobs:
       id-token: write # To sign the provenance.
       packages: write # To upload assets to release.
       actions: read #To read the workflow path.
-    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@9dc6318aedc3d24ede4e946966d30c752769a4f9
+    # NOTE: The container generator workflow is not officially released as GA.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.2.2
     with:
       image: ghcr.io/${{ github.repository_owner }}/kyverno
       digest: "${{ needs.push-kyverno.outputs.kyverno_sha256_digest }}"
       registry-username: ${{ github.actor }}
-    # TODO(https://github.com/slsa-framework/slsa-github-generator/issues/492): Remove after GA release.
-      compile-generator: true
     secrets:
       registry-password: ${{ secrets.CR_PAT }}
-      
+
   generate-kyverno-cli-provenance:
     needs: push-kyverno-cli
     permissions:
       id-token: write # To sign the provenance.
       packages: write # To upload assets to release.
       actions: read #To read the workflow path.
-    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@9dc6318aedc3d24ede4e946966d30c752769a4f9
+    # NOTE: The container generator workflow is not officially released as GA.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.2.2
```

Comment:
@ianlewis thanks for the PR, the pre-checks are failing which ensure the SHA pinned actions instead of version. 

```
  Error: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.2.2 is not pinned to a full length commit SHA.
  Error: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.2.2 is not pinned to a full length commit SHA.
  Error: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.2.2 is not pinned to a full length commit SHA.

```

Reply:
We don't recommend that users use a pinned SHA for our workflows due to limitations in GitHub Actions. We need to get information about the workflow that is running via an OIDC token which doesn't report branch or tag information.
https://github.com/slsa-framework/slsa-github-generator#referencing-slsa-builders-and-generators

Would it be ok to add an exception for this workflow in the pre-checks?

Reply:
@ianlewis - thanks for the PR!

>Would it be ok to add an exception for this workflow in the pre-checks?

This seems possible, by configuring `allowlist` in `.github/workflows/tests.yaml`:

https://github.com/zgosalvez/github-actions-ensure-sha-pinned-actions#inputs

@eddycharly @chipzoller - do you see any reason that we cannot do this?

Reply:
No, this should be fine to do. An exception here is warranted.

Reply:
Thanks. I'll update the PR.